### PR TITLE
Refactored actions to include deployment to the staging environment

### PR DESCRIPTION
**Pull request by longrunningprocess**
## Description

This PR will will trigger automated build, test, publish and deployment actions  for any pushes to `develop`

### Type of Change

- Engineering

## Screenshots

N/A

## How Has This Been Tested?

- [x] I originally had the trigger set up to run off changes in this branch.  I verified the full workflow as well as deployment to the staging environment.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have reviewed the title/description of this PR which will be used as the squashed PR commit message
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: .github/workflows/build-and-deploy-images.yml [hunk 1]
```diff
@@ -5,8 +5,6 @@ on:
   push:
     tags:
       - "v*"
-    branches:
-      - develop
```
Comment (megahirt) [inline .github/workflows/build-and-deploy-images.yml:9]: Just a note to say that we should remove the section regarding developer builds (lines 19-24).  We only every want to build production builds on github actions.

File: docker/Makefile [hunk 2]
```diff
@@ -24,14 +24,7 @@ endif
 .PHONY: unit-tests
 unit-tests: build
 	docker-compose build test-php
-ifeq ("$(TEAMCITY_VERSION)",$())
-	# developer machine
 	docker-compose run test-php
-else
-	# teamcity CI
-	docker-compose run -e WEBSITE=languageforge.org --name unittests test-php
-	docker cp unittests:/var/www/PhpUnitTests.xml ..
```
Comment (megahirt) [inline docker/Makefile:33]: note to self: this line is what we need to use for GA to recognize the PHP Unit test results in the test formatter

Comment (megahirt) [reply docker/Makefile:33]: There is also discussion about this here: https://github.com/sillsdev/web-languageforge/pull/1258#discussion_r760716331

**Review by megahirt — APPROVED** [review]
Looking good!

File: .github/workflows/build-and-deploy-images.yml (hunk 1 above)
Comment (longrunningprocess) [reply .github/workflows/build-and-deploy-images.yml:9]: got it, thanks!

**Pull request merged**